### ISSUE TITLE
Component with forwardRef passed as prop

### ISSUE THREAD

**Issue by tu4mo** [algolia/react-element-to-jsx-string#613]
```jsx
const A = () => <div />
const B = React.forwardRef(() => <div />)
```

What I get:

```jsx
<Comp
  a={A}
  b={{
    $$typeof: Symbol(react.forward_ref),
    render: _c3
  }}
/>
```

What I expect:

```jsx
<Comp
  a={A}
  b={B}
/>
```

Repro: https://codesandbox.io/s/dank-cdn-ftvik?file=/src/App.js

**Comment by armandabric** [comment]
Thanks to #617 we are now supporting `React.forwardRef()`. But I still reproduce your issue with this test:

```js
it('should use inferred function name as display name for `forwardRef` element when used in a props', () => {
    const Comp = () => <div />;
    const A = () => <div />;
    const B = React.forwardRef(function B(_props, ref) {
      return <div ref={ref} />;
    });

    expect(reactElementToJSXString(<Comp a={A} b={B}></Comp>)).toEqual(
      `<Comp
  a={function noRefCheck() {}}
  // Should be <B />
  b={{
    $$typeof: Symbol(react.forward_ref),
    render: function noRefCheck() {}
  }}
/>`
    );
  });

**Comment by Andarist** [comment]
It's an interesting case - this is not a valid React element so it doesn't go into this logic:
https://github.com/algolia/react-element-to-jsx-string/blob/2c5132e58f69414d7b67e24cd470594656261b7f/src/formatter/formatPropValue.js#L47-L54

A valid React element is a result of the `React.createElement` call (so basically the output of JSX: `b={<B/>}`). This is a component type~, like a class, function, result of the `React.memo` and more.

**Comment by armandabric** [comment]
Good point. I'm not sure what we could propose as in fact `B` is only an object parsed to a prop: not a react component.

**Comment by Andarist** [comment]
Well, it could still be formatted somehow - since we can recognize that it's not "just an object". But I don't believe printing `<Comp/>` is OK here. The suggested expected formatting could work OK here - printing `Comp` in this case. Note that in the given example we would not be able to print `B` because everything is anonymous there, but with inferred names and display names something nicer could be printed.

I probably won't take this work on myself though - it shouldn't be overly hard though with the groundwork being laid out already.

**Comment by armandabric** [comment]
I agree with you we could easily add a dedicated render for this use case but I lack of idea on the shape of this render. 

If people really need this they could explain us their use case and make us proposal. If it's a reasonable use case we will consider to handle this. I keep this issue open for the discussion.

**Comment by tu4mo** [comment]
My exact use case is documenting polymorphic components with `as` prop.

```jsx
<Link as={RouterLink}>Link</Link>
```

If the `as` component is wrapped with a forwardRef (like they usually are) it does not work.

**Comment by armandabric** [comment]
I see. We could maybe have a dedicated behavior for the `as` props. It's a common usage

**Comment by Andarist** [comment]
OTOH - this really could be handled in the same way for any prop. People probably rarely want to see the react element's internal structure in their stringified trees

**Comment by iMoses-Apiiro** [comment]
I think we have two cases here to handle:
- a prop which is of type `object` and has a `$$typeof` property of type `symbol`, in which case try to use `displayName`
- a prop which is of type `function` that has a `displayName`

I can come up with a quick code example but I'm not sure if these are the right places to put it:

```
  if (typeof propValue.$$typeof === 'symbol') {
    return `{${propValue.displayName || 'No Display Name'}}`;
  }

  if (typeof propValue === 'function') {
    return `{${propValue.displayName || formatFunction(propValue, options)}}`;
  }
```

**Comment by as-zlynn-philipps** [comment]
Is there a PR for this? I don't see one linked.
